app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -274,6 +304,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
     @Override
     protected void onResume() {
         super.onResume();
+
+        updateLaunchButtonText(Pokefly.isRunning() || pokeFlyStarting);
```

Comment:
1. Just send an "UPDATE_BUTTON" intent to ourselves instead of calling this function.